plutus-playground-server/usecases/Starter.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE DeriveAnyClass      #-}
+{-# LANGUAGE DeriveGeneric       #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE NoImplicitPrelude   #-}
+{-# LANGUAGE OverloadedStrings   #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TemplateHaskell     #-}
+{-# LANGUAGE TypeApplications    #-}
+{-# OPTIONS_GHC -fno-ignore-interface-pragmas #-}
+module Starter where
+-- TRIM TO HERE
+-- This is a starter contract, based on the Game contract,
+-- containing the bare minimum required scaffolding.
+--
+-- What you should change to something more suitable for
+-- your use case:
+--   * The DataScript type
+--   * The Redeemer type
+--
+-- And add function implementations (and rename them to
+-- something suitable) for the endpoints:
+--   * publish
+--   * redeem
+
+import qualified Language.PlutusTx          as PlutusTx
+import           Language.PlutusTx.Prelude
+import           Ledger                     (Address, DataScript, PendingTx,
+                                             RedeemerScript, ValidatorScript (ValidatorScript),
+                                             compileScript, scriptAddress)
+import           Ledger.Value               (Value)
+import           Playground.Contract
+import           Wallet                     (MonadWallet, WalletAPI, WalletDiagnostics, collectFromScript,
+                                             defaultSlotRange, payToScript_, startWatching)
+-- | The data script type.
+data ContractDataScript = ContractDataScript ByteString
+PlutusTx.makeLift ''ContractDataScript
+
+-- | Redeemer type.
+data Redeemer = Redeemer ByteString
+PlutusTx.makeLift ''Redeemer
+
+-- | This method is the spending validator (which gets lifted to
+--   its on-chain representation).
+validateSpend :: ContractDataScript -> Redeemer -> PendingTx -> Bool
+validateSpend _dataScript _redeemerScript _ = True
+
+-- | This function lifts the validator previously defined to
+--   the on-chain representation.
+contractValidator :: ValidatorScript
+contractValidator =
+    ValidatorScript ($$(Ledger.compileScript [|| validateSpend ||]))
+
+-- | Helper function used to build the DataScript.
+mkDataScript :: String -> DataScript
+mkDataScript = undefined
```

Comment:
`undefined` should be `error` with an appropriate message.

Also I'd prefer if the starter contract was actually runnable without making any changed. So maybe change the types of redeemer and datascript from `ByteString` to `()`, and use the correct `Ledger.lift` functions in `mkDataScript` and `mkRedeemerScript`

Reply:
I am using an `Integer` instead of `()` because of swagger's complaints about not being able to encode `()`.

Reply:
:thinking: that seems weird

Reply:
@michaelpj what seems weird? The fact that swagger does not know how to encode `()` by default, or the fact that I am using an `Integer` instead of `()`? Or both?

Reply:
The fact that swagger doesn't know how to encode it. Do we not have units in other places? Perhaps they don't appear in endpoints?

Reply:
Yes. It does not make much sense for an endpoint to send `()`, but I would still like it to be there to emphasise that the developer will need to put something there.

Reply:
I could add `()` support and just encode it as `null`. Not sure what the best way to represent it in the UI is though. 

Would it not make more sense to put a type hole there though?

(FWIW It's actually not Swagger any more.)